File: Practice/src/coding_basic_training/ManipulateNumbers2.js
function solution(numLog) {
    var answer = "";
    for (let i = 1; i < numLog.length; i++) {
        switch (numLog[i] - numLog[i - 1]) {
            case 1:
                answer += "w";
                break;
            case -1:
                answer += "s";
                break;
            case -10:
                answer += "a";
                break;
            case 10:
                answer += "d";
        }
    }
    return answer;
    // 객체를 이용할 수 있음
    // let a = {
    //     1: "w",
    //     "-1": "s",
    //     10: "d",
    //     "-10": "a",
    // };

    // return numLog
    //     .slice(1)
    //     .map((v, i) => {
    //         return a[v - numLog[i]];
    //     })
    //     .join("");
}

File: Practice/src/coding_basic_training/ManipulateNumbers2.java
package coding_basic_training;

public class ManipulateNumbers2 {
//	수 조작하기2
	public String solution(int[] numLog) {
		String answer = "";
		for (int i = 1; i < numLog.length; i++) {
			switch (numLog[i] - numLog[i - 1]) {
			case 1:
				answer += 'w';
				break;
			case -1:
				answer += 's';
				break;
			case -10:
				answer += 'a';
				break;
			case 10:
				answer += 'd';
			}
		}
		return answer;
	}
}